backends/utils/history_server.go

```
@@ -0,0 +1,617 @@
+	}
+
+	m.LogsContainer = defaultLogsContainer
+	if m.config.LogsContainer != "" {
+		m.LogsContainer = m.config.LogsContainer
+	}
+
+	m.MaxBytesInNginxRequest = defaultMaxBytesInNginxRequest
+	if m.config.MaxBytesInNginxRequest != 0 {
+		m.MaxBytesInNginxRequest = m.config.MaxBytesInNginxRequest
+	}
+
+	m.HistoryFileDurationSecondSpans = defaultHistoryFileDurationInSeconds
+	if m.config.HistoryFileDurationHourSpans != 0 {
+		m.HistoryFileDurationSecondSpans = m.config.HistoryFileDurationHourSpans * 3600
+	}
+
+	m.HistoryFileNum = defaultHistoryFileNum
+	if m.config.HistoryFileNum != 0 {
+		m.HistoryFileNum = m.config.HistoryFileNum
+	}
+}
+
+func (m *HistoryServer) getCurrentLogFileName() string {
+	// Round current time to the nearest log time-bucket
+	currentTimeInSeconds := time.Now().Unix()
+	currentTimeBucket := m.HistoryFileDurationSecondSpans * (currentTimeInSeconds / m.HistoryFileDurationSecondSpans)
+	return fmt.Sprintf("%v/%v.json", m.LogsFolderPath, currentTimeBucket)
+}
+
+func (m *HistoryServer) getLogFileNameByTime(timeBucket int64) string {
+	return fmt.Sprintf("%v/%v.json", m.LogsFolderPath, timeBucket)
+}
+
+func (m *HistoryServer) Start() {
+	go func() {
+		var pendingLogs []HistoryEntry
+
+		for {
+			select {
+			case entry := <-m.requests:
+				pendingLogs = append(pendingLogs, entry)
+				m.logs = append(m.logs, entry)
+
+				if len(pendingLogs) == m.PendingLogsBatchSize {
+					m.writeMonitoringBatch(pendingLogs)
+					pendingLogs = pendingLogs[:0]
+				}
+			case <-time.After(m.WriteMonitoringLogsTimeout):
+				if len(pendingLogs) > 0 {
+					m.writeMonitoringBatch(pendingLogs)
+					pendingLogs = pendingLogs[:0]
+				}
+			}
+		}
+	}()
+}
+
+func (m *HistoryServer) createDefaultV3ioClient() {
+
+	session := &frames.Session{}
+
+	token := "99b630dd-bcec-4d3f-a77e-31a3761cef28" // os.Getenv("V3IO_ACCESS_KEY")
+	if token == "" {
+		m.logger.Error("can not create v3io.client. could not find `V3IO_ACCESS_KEY` environment variable")
```

I still think you should return the error and not just log it here.

---

hmm i would still rather frames work without logging history, than not work at all. no?

---

You can still log it and move on eventually, but the way it's currently written you can't fail it even if you wanted to.